a8c-files.php

```diff
@@ -1156,6 +1156,59 @@ function a8c_files_maybe_inject_image_sizes( $data, $attachment_id ) {
 
 	return $data;
 }
+/**
+ * Filter dimensions for images to return a proper image size and srcset values
+ *
+ * @param bool   $value         The filtered value, defaults to `true`.
+ * @param string $image         The HTML `img` tag where the attribute should be added.
+ * @param string $context       Additional context about how the function was called or where the img tag is.
+ * @param int    $attachment_id The image attachment ID.
+ */
+function a8c_files_maybe_inject_dimensions( $value, $image, $context, $attachment_id ) {
+	$vipgo_img_src = preg_match( '/src="([^"]+)"/', $image, $match_src ) ? $match_src[1] : '';
+
+	/** This filter is documented in wp-admin/includes/media.php */
+	add_filter( 'wp_image_src_get_dimensions', function( $dimensions, $image_src, $image_meta, $attachment_id ) use ( $vipgo_img_src ){
+		$url = wp_parse_url( $vipgo_img_src );
+		if ( isset( $url['query'] ) ) {
+			parse_str( htmlspecialchars_decode($url['query']), $query_args );
+		}
+
+		// Let's calculate the aspect ratio of the image:
+		$aspect_ratio =  $image_meta['height'] / $image_meta['width'];
+
+		// Is it a Full Size image?  We're checking for sizes in query arguments
+		if ( 
+			! isset( $url['query'] ) &&
+			strpos( $image_src, $image_meta['file'] ) !== false
+		) {
+			$dimensions = array(
+				(int) $image_meta['width'],
+				(int) $image_meta['height'],
+		);
+		// If it's not Full Size, let's use our calculated Aspect Ratio to output proper dimensions according to the size
+		} else if ( 
```

Comment:
```suggestion
		} elseif ( 
```

Reply:
I will make the change but I'm curious as to why.  From the PHP docs it looks like `else if` and `elseif` are the same when using curly brackets but not the same when using colons.  Is it just a stylistic choice in this case?

Reply:
@mikeyarce It's a WPCS choice: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#use-elseif-not-else-if plus, `elseif` is then a unique keyword, should one need to find instances of it.